dht/token.hh

```diff
@@ -34,18 +34,20 @@ class token {
     // If a token with value INT64_MIN is generated by the hashing algorithm,
     // the result is coerced into INT64_MAX.
     // (So INT64_MAX is twice as likely as every other token.)
-    static inline int64_t normalize(int64_t t) {
+    static constexpr int64_t normalize(int64_t t) {
         return t == std::numeric_limits<int64_t>::min() ? std::numeric_limits<int64_t>::max() : t;
```

Comment:
We do:
```
bhalevy@lt scylla$ git grep -En '\[\[(un)?likely\]\]' | wc -l
95
```

But we use it sparingly to optimize common paths vs. exceptional paths that are rare by design.
But otherwise, for things that are workload-dependent, we would let branch prediction do its thing.

Reply:
@mykaul Why are you asking about `likely` in this context?

Reply:
"If a token with value INT64_MIN is generated" made me wonder how common is this scenario.

Reply:
> "If a token with value INT64_MIN is generated" made me wonder how common is this scenario.

With vnodes it would be due to hashing and so very rare.
With tablets, there is no such chance since we prepare the tablets token ranges starting from `dht::first_token()` which has the value of `min() + 1`